kernel/src/process.rs

```diff
@@ -1387,6 +1405,14 @@ impl<C: Chip> ProcessType for Process<'_, C> {
         }
     }
 
+    fn yield_wait_field(&self, addr: *mut u8) -> Option <&mut u8> {
+        if self.in_app_owned_memory(addr, 1) {
+            unsafe {Some(&mut *addr)}
+        } else {
+            None
+        }
+    }
```

Comment:
I think rather than using unsafe to return a reference, this would be better as `fn set_yield_flag(&self, addr: *mut u8, value: u8)`. Then accessing process memory is encapsulated in process.rs (which could be useful for host side testing and different Process implementations).

Reply:
agreed. This function is not currently marked `unsafe`, but it would be easy to create two mutable references to the same memory location by calling this twice, which is UB.